www/strf03/cv08/signin.php

```diff
@@ -0,0 +1,57 @@
+<?php
+session_start();
+require 'db.php';
+$submittedForm = $_SERVER['REQUEST_METHOD'] == 'POST';
+if ($submittedForm) {
+    $email = $_POST['email'];
+    $password = $_POST['password'];
+
+    $stmt = $db->prepare('SELECT * FROM users WHERE email = :email LIMIT 1');
+    $stmt->execute([
+        'email' => $email,
+    ]);
+
+    $existing_user = @$stmt->fetchAll()[0];
+    if (password_verify($password, $existing_user['password'])) {
+        $_SESSION['user_id'] = $existing_user['id'];
+        $_SESSION['user_email'] = $existing_user['email'];
+        $_SESSION['user_privilege'] = $existing_user['privilege'];
```

Comment:
tady mě jen napadla taková věc: ukládání info jako privilege do sessionu po prihlášení je na zvážení. takto by znamenalo, že  uživatel se musí znovu přihlásit po každé změně oprávnění. bylo by impozantnější kdyby on to nemusel dělat, tj. změníte hodnotu oprávnění v DB a rovnou může pokračovat v práci

Reply:
Taky jsem na ten problém narazil. Příště to tedy udělám jinak.